merino-suggest-traits/src/lib.rs

```diff
@@ -360,6 +369,72 @@ impl SupportedLanguages {
     }
 }
 
+/// A basic provider that never returns any suggestions.
+pub struct NullProvider;
+
+#[async_trait]
+impl SuggestionProvider for NullProvider {
+    fn name(&self) -> String {
+        "NullProvider".into()
+    }
+
+    fn cache_inputs(&self, _req: &SuggestionRequest, _hasher: &mut dyn CacheInputs) {
+        // No property of req will change the response
+    }
+
+    fn is_null(&self) -> bool {
+        true
+    }
+
+    async fn suggest(&self, _query: SuggestionRequest) -> Result<SuggestionResponse, SuggestError> {
+        Ok(SuggestionResponse::new(vec![]))
+    }
+
+    async fn reconfigure(
+        &mut self,
+        new_config: serde_json::Value,
+        _make_fresh: &MakeFreshType,
+    ) -> Result<(), SetupError> {
+        // make sure this is the right kind of config
+        convert_config(new_config)
+    }
+}
+
+/// Convert a [`serde_json::Value`] into an appropriate provider configuration object.
+/// # Errors
+/// If the `Value` passed does not conform to the expected schema of the config,
+/// a `SetupError::InvalidConfiguration` error will be returned.
+pub fn convert_config<T: DeserializeOwned>(config: serde_json::Value) -> Result<T, SetupError> {
+    serde_json::from_value::<T>(config)
+        .context("loading provider config")
+        .map_err(SetupError::InvalidConfiguration)
+}
+
+/// Reconfigure the passed provider, or if there is a problem reconfiguring it,
+/// replace it with a newly created provider.
```

Comment:
Can you note here that if `make_fresh` also fails, the original provider will remain unchanged?